pattoo_shared/installation/configure.py

```diff
@@ -0,0 +1,313 @@
+                'ip_bind_port': 20201
+            },
+            'pattoo_web_api': {
+                'ip_address': '127.0.0.1',
+                'ip_bind_port': 20202,
+            }
+        }
+
+        self.default_server_config = {
+            'pattoo_db': {
+                'db_pool_size': 10,
+                'db_max_overflow': 20,
+                'db_hostname': 'localhost',
+                'db_username': 'pattoo',
+                'db_password': 'password',
+                'db_name': 'pattoo'
+            },
+            'pattoo_api_agentd': {
+                'ip_listen_address': '0.0.0.0',
+                'ip_bind_port': 20201,
+            },
+            'pattoo_apid': {
+                'ip_listen_address': '0.0.0.0',
+                'ip_bind_port': 20202,
+            },
+            'pattoo_ingesterd': {
+                'ingester_interval': 3600,
+                'batch_size': 500,
+                'graceful_timeout': 10
+            }
+        }
+
+    def create_user(self, user_name, directory, shell, verbose):
+        """Create user and their respective group.
+
+        Args:
+            user_name: The name and group of the user being created
+            directory: The home directory of the user
+            shell: The shell of the user
+            verbose: A boolean value that a allows the script to run in verbose
+            mode
+
+        Returns:
+            None
+        """
+        # Ensure user has sudo privileges
+        if getpass.getuser() == 'root':
+
+            # If the group specified does not exist, it gets created
+            if not self.group_exists(user_name):
+                shared.run_script('groupadd {0}'.format(user_name), verbose)
+
+            # If the user specified does not exist, they get created
+            if not self.user_exists(user_name):
+                shared.run_script(
+                    'useradd -d {1} -s {2} -g {0} {0}'.format(
+                        user_name, directory, shell), verbose)
+        else:
+            # Die if not root
+            shared.log('You are currently not running the script as root')
+
+    def group_exists(self, group_name):
+        """Check if the group already exists.
+
+        Args:
+            group_name: The name of the group
+
+        Returns
+            True if the group exists and False if it does not
+        """
+        try:
+            # Gets group name
+            grp.getgrnam(group_name)
+            return True
+        except KeyError:
+            return False
+
+    def user_exists(self, user_name):
+        """Check if the user already exists.
+
+        Args:
+            user_name: The name of the user
+
+        Returns
+            True if the user exists and False if it does not
+
+        """
+        try:
+            # Gets user name
+            pwd.getpwnam(user_name)
+            return True
+        except KeyError:
+            return False
+
+    def pattoo_config(
```

Comment:
This should be a function not a method. It uses no instance variables.

Reply:
At least, only uses two, but why also pass arguments?

Reply:
Ideally, you should pass two variables. The default dict, and the dict read in from the configuration directory. The function should return the result.

Reply:
I just went back through the code and realized that read_config goes through the default config file and returns the result based on the dictionary passed in, so this function only makes the directories and creates the configuration file.
Would it still be necessary to return the result?